spec/presenter_spec.rb

```diff
@@ -13,55 +13,110 @@ def display(message)
     end
   end
 
-  class FakeFormatter
-    def format(*)
+  class StubFormatter
+    def format(board)
+      "|#{board}|"
+    end
+  end
+
+  class StubMessages
+    def winning(player)
+      "#{player} won"
+    end
+
+    def tied
+      'tied'
+    end
+
+    def prompt(player)
+      "#{player} prompt"
+    end
+  end
+
+  class BoardGame
+    def board
       'board'
     end
   end
 
-  class GetOnlyBoard
-    attr_reader :size
+  class WonGame < BoardGame
+    def win?
+      true
+    end
 
-    def initialize(size)
-      @size = size
+    def tie?
+      false
     end
+  end
 
-    def get(at:)
-      at
+  class TieGame < BoardGame
+    def win?
+      false
+    end
+
+    def tie?
+      true
     end
   end
 
-  def default_presenter(displayer)
-    Presenter.new(
-      displayer: displayer,
-      clear: clear,
-      formatter: FakeFormatter.new
-    )
+  class ContinueGame < BoardGame
+    def win?
+      false
+    end
+
+    def tie?
+      false
+    end
   end
 
-  let(:clear) { "\e[H\e[2J" }
+  class StubPlayer
+    def symbol
+      'X'
+    end
+  end
 
   let(:displayer) { DisplayLogger.new }
 
-  let(:board) { GetOnlyBoard.new(3) }
+  let(:clear) { 'clear' }
 
-  let(:test_message) { 'message' }
+  let(:player) { StubPlayer.new }
+
+  def default_presenter(displayer)
+    Presenter.new(
+      displayer: displayer,
+      clear: clear,
+      formatter: StubFormatter.new,
+      messages: StubMessages.new
+    )
+  end
 
   it 'clears the display' do
     presenter = default_presenter(displayer)
```

Comment:
I remember something about it being dangerous to have let depend on other let's. As of right now I need access to that displayer within the tests, and it is a let. So would making presenter into a let be an issue?

Reply:
I've seen lets used with other lets–one trick of this to have less duplication and have contexts define a let used in the parent let. Because of the lazy loading it works.

Reply:
Got it, going to look into contexts! Thank you! :)

Reply:
Sometimes adding a helper function to the spec can encapsulate setup and reduce the need to have lets depending on lets. Not sure if that applies here or not, just wanted to mention it.